tests/mock_crust/mod.rs

```diff
@@ -177,15 +180,18 @@ fn joining_node_with_ignoring_candidate_info() {
     let network = Network::new(MIN_SECTION_SIZE, None);
     let mut nodes = create_connected_nodes(&network, MIN_SECTION_SIZE);
     let bootstrap_config = BootstrapConfig::with_contacts(&[nodes[0].handle.endpoint()]);
-    for node in nodes.iter_mut() {
-        node.inner.set_ignore_candidate_info_counter(1);
+    // Make half of the elders ingoring the candidate_info to simulate lagging.
+    // Without the resending, this will block the joining node to be approved.
+    for i in 0..((nodes.len() + 1) / 2) {
```

Comment:
for node in nodes.iter_mut().take(2) {
...
}

Reply:
hmm... `2` only works for the `min_network_size` which is 4. 
I thought we want a more general pattern?
how about: `nodes.iter_mut().take((nodes.len() + 1) / 2)` ?

Reply:
Ok with your suggestion, but why the +1?
Why not `take(nodes.len() / 2)`?

Reply:
for `odd` numbers, `5 / 2 = 2`, which is `less than half`
maybe it doesn't matter, as we require supermajority.
I will use `nodes.len() / 2` then.